scripts/profiler/processed_events.py

```diff
@@ -32,24 +32,24 @@ def match_event_processing(self):
             'event_processing_start')
         self.event_processing_end_id = self.raw_data.get_event_type_id(
             'event_processing_end')
-
-        if (self.event_processing_start_id is None) or \
-          (self.event_processing_end_id is None):
-            self.tracking_execution = False
-            for ev in self.raw_data.events:
-                self.tracked_events.append(TrackedEvent(ev, None, None))
-            return
-
-        for i in range(1, len(self.raw_data.events)):
+        for i in range(len(self.raw_data.events)):
```

Comment:
why not iterate over `self.raw_data.events`

btw - another thing I noticed at line 44... do I correctly see that we are traversing list for every event seen in the past :(
why `dict` was not used? We could add submitted event to `dict` with address used as key and identify it back when we get start/end execution. There is a question where to throw the accumulated events but I think it could be resolved and end result would be more optimal.

Reply:
`i` is used as an argument in line 44. So if I were to iterate over `self.raw_data.events` I would have to either keep count of `i` or use `index` function which require `for` loop.

Reply:
Notice that in line 44 we are iterating backward and in most cases "event processing start" is sent right after event so this loop doesn't require much iterations. Another thing is, that memory address is not unique. Once an event with its processing start and processing end is sent, memory addressed is released and later reused, so it would be required to delete events from dictionary once they are processed (which is not a big deal but I'm pointing some problems regarding this solution).

Reply:
Anyway, in another profiler rework matching event is done differently. I will keep an idea with dictionary in mind and examine whether it would be useful there.